src/amplitude-client.js

```diff
@@ -76,17 +78,37 @@ AmplitudeClient.prototype.init = function init(apiKey, opt_userId, opt_config, o
   }
 
   try {
+    _parseConfig(this.options, opt_config);
+
+    if (this.options.cookieName !== DEFAULT_OPTIONS.cookieName) {
+      utils.log.warn('The cookieName option is deprecated. We will be ignoring it for newer cookies');
+    }
+
     this.options.apiKey = apiKey;
-    this._storageSuffix = '_' + apiKey + this._legacyStorageSuffix;
+    this._storageSuffix = '_' + apiKey + (this._instanceName === Constants.DEFAULT_INSTANCE ? '' : '_' + this._instanceName);
+    this._storageSuffixV5 = apiKey.slice(0,6);
+
+    this._oldCookiename = this.options.cookieName + this._storageSuffix;
+    this._unsentKey = this.options.unsentKey + this._storageSuffix;
+    this._unsentIdentifyKey = this.options.unsentIdentifyKey + this._storageSuffix;
+
+    this._cookieName = COOKIE_PREFIX + this._storageSuffixV5;
+
+    this._metadataStorage = new MetadataStorage({
+      storageKey: this._cookieName,
+      disableCookies: this.options.disableCookies
+    });
+
+    const hasOldCookie = !!this.cookieStorage.get(this._oldCookiename);
+    const hasNewCookie = !!this._metadataStorage.load();
+    this._useOldCookie = !hasNewCookie && hasOldCookie;
+    const hasCookie = hasNewCookie || hasOldCookie;
```

Comment:
should we just migrate the old cookie data into the new metastorage so you don't need to juggle between the two? what's the benefit of keeping the old one around?

Reply:
@djih, If someone is relying on the same cookie in multiple products, they would need to be sure to update all cookies at the same time. Otherwise the different products would be out of sync. This seemed like the safest approach.